src/kagglehub/clients.py

```diff
@@ -198,6 +201,24 @@ def download_file(
                         _CHECKSUM_MISMATCH_MSG_TEMPLATE.format(expected_md5_hash, actual_md5_hash)
                     )
 
+            # For individual file downloads, the downloaded file may be a zip of the file rather
```

Comment:
Can you split the bugfix to its own PR? Can you also include a unit test for the fix?